modules/phase_field/src/kernels/EpsilonModelKernel2ndGauss.C

```diff
@@ -0,0 +1,41 @@
+
+#include "EpsilonModelKernel2ndGauss.h"
+
+registerMooseObject("PhaseFieldApp", EpsilonModelKernel2ndGauss);
+
+InputParameters
+EpsilonModelKernel2ndGauss::validParams()
+{
+  InputParameters params = ADKernel::validParams();
+  params.addClassDescription("Implements the term:-L\\nabla \\cdot \\left( \\frac{\\partial "
+                             "m(\\theta, v)}{\\partial \\nabla\\eta_i} f_0 \\right)");
+
+  params.addParam<MaterialPropertyName>("f_name", "F", " Base name of the free energy function");
+
+  return params;
+}
+
+EpsilonModelKernel2ndGauss::EpsilonModelKernel2ndGauss(const InputParameters & parameters)
+  : ADKernel(parameters),
+    _L_AD(getADMaterialProperty<Real>("L_AD")),
+
+    _dmdx(getADMaterialProperty<Real>("dmdx")),
+    _dmdy(getADMaterialProperty<Real>("dmdy")),
+    _dmdz(getADMaterialProperty<Real>("dmdz")),
+
+    _F(getMaterialProperty<Real>("F"))
+
+{
+}
+
+ADReal
+EpsilonModelKernel2ndGauss::computeQpResidual()
+{
+
+  ADReal gradtestx = (_grad_test[_i][_qp](0));
+  ADReal gradtesty = (_grad_test[_i][_qp](1));
+  ADReal gradtestz = (_grad_test[_i][_qp](2));
+
+  return _L_AD[_qp] * (_F[_qp]) *
+         (((_dmdx[_qp]) * gradtestx) + ((_dmdy[_qp]) * gradtesty) + ((_dmdz[_qp]) * gradtestz));
```

Comment:
```suggestion
    _dmdx(getADMaterialProperty<Real>("dmdx")),
    _dmdy(getADMaterialProperty<Real>("dmdy")),
    _dmdz(getADMaterialProperty<Real>("dmdz")),
    _F(getMaterialProperty<Real>("F"))
{
}

ADReal
EpsilonModelKernel2ndGauss::computeQpResidual()
{
  const auto & ADReal gradtest = _grad_test[_i][_qp];

  return _L_AD[_qp] * (_F[_qp]) *
         (_dmdx[_qp] * gradtest(0) + _dmdy[_qp] * gradtest(1) + _dmdz[_qp] * gradtest(2));
```

And so on. Or even better build a TypreVector (RealGradient) out of _dmdx, _dmdy, _dmdz and use a simople scalar product instead of messing around with the components.

Reply:
That was fixed. See updated files.